### PR TITLE
docs: create `metadata.rst`

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
Move these items to a separate page.

## How I Tested These Changes
local

### CODE REVIEW

**Comment by rexledesma** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21172?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#21172** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21172?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#21170** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21170?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21169** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21169?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rexledesma and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/sphinx/sections/api/apidocs/metadata.rst [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+.. autoclass:: TextMetadataValue
+
+.. autoclass:: TimestampMetadataValue
+
+.. autoclass:: UrlMetadataValue
+
+Tables
+^^^^^^
+
+These APIs provide the ability to express column schemas (`TableSchema`), rows/records (`TableRecord`), and column lineage (`TableColumnLineage`) in Dagster.
```
Comment (erinkcochran87) [inline docs/sphinx/sections/api/apidocs/metadata.rst:51]: - Should we be specific and say `... and column lineage in Dagster as metadata`?
- Do we need to specify anything about ops and/or assets? These were originally on the ops page, so not completely sure.

**Review by erinkcochran87 — APPROVED** [review]

File: docs/sphinx/sections/api/apidocs/metadata.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
+.. currentmodule:: dagster
+
+Metadata
+========
+
+Dagster uses metadata to communicate arbitrary user-specified metadata about structured
```
Comment (erinkcochran87) [inline docs/sphinx/sections/api/apidocs/metadata.rst:6]: Perhaps link to the metadata concept page here: `/concepts/metadata-tagging`

File: docs/sphinx/sections/api/apidocs/metadata.rst (hunk 1 above)
Comment (rexledesma) [reply docs/sphinx/sections/api/apidocs/metadata.rst]: I think the UI portion should be mentioned in the concepts page, rather than the API docs.

**Pull request merged**